PocketKit/Sources/PocketKit/PocketNotificationService.swift

```diff
@@ -0,0 +1,170 @@
+import Foundation
+import BrazeKit
+import BrazeUI
+import Sync
+import Analytics
+import SharedPocketKit
+import Combine
+import UIKit
+import UserNotifications
+
+class PocketNotificationService: NSObject {
+    
+    /**
+     The Braze service object
+     */
+    private let braze: Braze
+    
+    /**
+     Instance of our API client for pulling in data when we recieve pushes.
+     */
+    private let source: Source
+    
+    /**
+     Instance of the Pocket Analytics tracker
+     */
+    private let tracker: Tracker
+    
+    /**
+     Instance of the Pocket Session manager for us to subscribe to
+     */
+    private let sessionManager: AppSession
+    
+    /**
+     Subscriptions that this object listens to.
+     */
+    private var subscriptions: Set<AnyCancellable> = []
+    
+    
+    init(source: Source, tracker: Tracker, sessionManager: AppSession) {
+        self.source = source
+        self.tracker = tracker
+        self.sessionManager = sessionManager
+        
+        // Init Braze with our information.
+        var configuration = Braze.Configuration(
+            apiKey: Keys.shared.brazeAPIKey,
+            endpoint: Keys.shared.brazeAPIEndpoint
+        )
+        
+        // Enable logging of general SDK information (e.g. user changes, etc.)
+        configuration.logger.level = .info
+        configuration.push.appGroup = "group.com.ideashower.ReadItLaterProAlp"
+        braze = Braze(configuration: configuration)
+        super.init()
+        
+        // Set up the in app message ui
+        let inAppMessageUI = BrazeInAppMessageUI()
+        inAppMessageUI.delegate = self
+        self.braze.inAppMessagePresenter = inAppMessageUI
+        
+        sessionManager.$currentSession.receive(on: DispatchQueue.main).sink { [weak self] session in
+            if let session = session {
```

Comment:
I think you'll see that a lot of swift code would prefer a [guard statement][guard] here to bail out of the function if proper conditions aren't met.

e.g.:

```swift
func foo(bar: String) {
  // make sure all our state and/or inputs are valid before proceeding
  guard !bar.isEmpty else { return }
  doSomething(with: bar)
}
```

The effect is that we eliminate "right drift" and also the main thrust of the function's work is a bit more apparent.

[guard]: https://docs.swift.org/swift-book/ReferenceManual/Statements.html#ID524